packages/noya-state/src/reducers/applicationReducer.ts

```diff
@@ -218,6 +237,16 @@ export function applicationReducer(
 
             style.value = styleReducer(style.value, action);
 
+            if (
+              action[0] === 'setFillFillType' &&
+              action[2] === Sketch.FillType.Pattern &&
+              style.value.fills
+            ) {
+              const _ref = style.value.fills[action[1]].image?._ref;
+              if (!_ref) return;
+              draft.sketch.images[_ref] = CHECKERED_BACKGROUND_BYTES;
```

Comment:
Should this pattern initialization the new code above? Maybe we should make a function so we don't have it duplicated twice

Reply:
which code ? 🤔

Reply:
oh I understand !